test/unit/StakingProducts/recalculateEffectiveWeight.js

```diff
@@ -0,0 +1,447 @@
+const { ethers } = require('hardhat');
+const { expect } = require('chai');
+
+const { parseEther } = ethers.utils;
+const { Zero, One } = ethers.constants;
+
+const {
+  allocateCapacity,
+  depositTo,
+  burnStake,
+  setStakedProducts,
+  daysToSeconds,
+  burnStakeParams,
+  buyCoverParamsTemplate,
+  newProductTemplate,
+} = require('./helpers');
+const { setEtherBalance } = require('../../utils/evm');
+const { increaseTime } = require('../../utils').evm;
+
+const DEFAULT_PRODUCTS = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20];
+describe('recalculateEffectiveWeight', function () {
+  it('should return 0 for products not found in stakingPool', async function () {
+    const { stakingProducts } = this;
+    const productIdToAdd = Zero;
+
+    await setStakedProducts.call(this, { productIds: [productIdToAdd] });
+
+    const stakedProduct = await stakingProducts.getProduct(this.poolId, productIdToAdd);
+    expect(stakedProduct.lastEffectiveWeight).to.be.equal(100);
+    expect(stakedProduct.targetWeight).to.be.equal(100);
+
+    const unknownProduct = await stakingProducts.getProduct(this.poolId, productIdToAdd.add(1));
+    expect(unknownProduct.lastEffectiveWeight).to.be.equal(0);
+
+    expect(await stakingProducts.getTotalTargetWeight(this.poolId)).to.be.equal(100);
+    expect(await stakingProducts.getTotalEffectiveWeight(this.poolId)).to.be.equal(100);
+
+    // recalculating should do nothing
```

Comment:
Something that I highlighted a few times so far in these tests is to test what it says in the it text and not add extra assertions 

for example this verification that  `/ recalculating should do nothing`

is a different verification than

`should return 0 for products not found in stakingPool`

While it's nice to verify the unit test does more than it's advertised to do.

That property should be a different test imo.

This level of granularity is specific to unit tests only, in the integration tests it's customary to assert many things in one block.